py3status/modules/diskdata.py

```diff
@@ -4,22 +4,27 @@
 
 Configuration parameters:
     cache_timeout: refresh interval for this module. (default 10)
-    disk: show stats for disk or partition, i.e. `sda1`. None for all disks.
+    disk: show stats for mountpoint, disk or partition, i.e. `sda1`.
+        None for all disks, filtered by `type` parameter.
```

Comment:
I choose to stick on a string so the change isn't breaking for current users. If a breaking change is ok, using list is a better option for sure. You're right about the documentation, I'll integrate on current choice for now

Reply:
>isn't breaking for current users.

I think add this and we're good here. `isinstance(config, str)` not supported in Python2. 
```diff
diff --git a/py3status/modules/diskdata.py b/py3status/modules/diskdata.py
index 08ef8a5..4271f95 100644
--- a/py3status/modules/diskdata.py
+++ b/py3status/modules/diskdata.py
@@ -87,6 +87,12 @@ class Py3status:
             value - value (float)
             unit - unit (string)
         """
+        # deprecation
+        if self.disk is None:
+            self.disk = []
+        elif not isinstance(self.disk, list):
+            self.disk = [self.disk]
+
         self.last_diskstats = self._get_diskstats(self.disk)
         self.last_time = time()
 
```

Reply:
LGTM, pushed in second revision

Reply:
Omit `split()` too.

Reply:
Here's the whole deprecation. I rename it to `disks` too because that makes sense now too.
```diff
diff --git a/py3status/modules/diskdata.py b/py3status/modules/diskdata.py
index 3191269..3b74af6 100644
--- a/py3status/modules/diskdata.py
+++ b/py3status/modules/diskdata.py
@@ -7,7 +7,7 @@ Configuration parameters:
     disk: mountpoints, disks or partitions list to show stats
         for, i.e. `["sda1", "/home", "/dev/sdd"]`.
         None for all disks, filtered by `type` parameter.
-        (default None)
+        (default [])
     format: display format for this module.
         (default "{disk}: {used_percent}% ({total})")
     format_rate: display format for rates value
@@ -71,7 +71,7 @@ class Py3status:
 
     # available configuration parameters
     cache_timeout = 10
-    disk = None
+    disks = []
     format = "{disk}: {used_percent}% ({total})"
     format_rate = "[\?min_length=11 {value:.1f} {unit}]"
     format_space = "[\?min_length=5 {value:.1f}]"
@@ -85,7 +85,16 @@ class Py3status:
     unit = "B/s"
 
     class Meta:
-        deprecated = {"remove": [{"param": "sector_size", "msg": "obsolete parameter"}]}
+        deprecated = {
+            "remove": [{"param": "sector_size", "msg": "obsolete parameter"}],
+            "rename": [
+                {
+                    "param": "disk",
+                    "new": "disks",
+                    "msg": "rename parameter to `disks`",
+                },
+            ],
+        }
 
     def _resolve_disk_name(self, path):
         disk_path = os.path.realpath(path)
@@ -105,10 +114,10 @@ class Py3status:
             unit - unit (string)
         """
         # deprecation
-        if self.disk is None:
-            self.disk = []
-        elif not isinstance(self.disk, list):
-            self.disk = [self.disk]
+        if self.disks is None:
+            self.disks = []
+        elif not isinstance(self.disks, list):
+            self.disks = [self.disks]
         types_info = {
             "disks": {"major": [3, 8, 22]},
             "raid": {"major": [9]},
@@ -138,7 +147,7 @@ class Py3status:
 
         self._disks = {}
 
-        if not self.disk:
+        if not self.disks:
             allowed_types = set()
             for _type in self.types:
                 allowed_types.update(types_info[_type]["major"])
@@ -148,7 +157,7 @@ class Py3status:
                     if int(line[0]) in allowed_types and int(line[1]) == 0:
                         self._add_monitored_disk("/dev/" + line[2])
         else:
-            for disk in self.disk.split():
+            for disk in self.disks:
                 # mountpoint or fully qualified device path
                 if disk.startswith("/"):
                     # mountpoint
```

Reply:
done